### PR TITLE
Base type controller

### PR DESCRIPTION
**Pull request by sandervankasteel**
# BE WARNED! This is untested and experimental code!!!!!

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -14,8 +14,12 @@
         "mockery/mockery": "0.9.*",
         "phpunit/phpunit": "~5.0",
         "symfony/css-selector": "3.1.*",
+<<<<<<< HEAD
```
Comment (mro95) [inline composer.json]: @Darkwater NUUUUU

Comment (darkwater) [inline composer.json]: fix pls

**Review by darkwater — CHANGES_REQUESTED** [review]
Get that merge conflict stuff outta here

**Comment by sandervankasteel** [comment]
Merge conflict has been resolved

**Review by darkwater — APPROVED** [review]

**Pull request merged**